src/ffi/python/mod.rs

```diff
@@ -153,19 +153,43 @@ fn init() -> PyResult<Driver> {
 	Ok(Driver(Some(rt_stop_tx)))
 }
 
-#[pyfunction]
-fn get_default_config() -> crate::api::Config {
-	let mut conf = crate::api::Config::new("".to_string(), "".to_string());
-	conf.host = Some(conf.host().to_string());
-	conf.port = Some(conf.port());
-	conf.tls = Some(false);
-	conf
+#[pymethods]
+impl User {
+	#[getter]
+	fn get_id(&self) -> pyo3::PyResult<String> {
+		Ok(self.id.to_string())
+	}
+
+	#[setter]
+	fn set_id(&mut self, value: String) -> pyo3::PyResult<()> {
+		self.id = value
+			.parse()
+			.map_err(|x: <uuid::Uuid as std::str::FromStr>::Err| {
+				pyo3::exceptions::PyRuntimeError::new_err(x.to_string())
+			})?;
+		Ok(())
+	}
+
+	#[getter]
+	fn get_name(&self) -> pyo3::PyResult<String> {
+		Ok(self.name.clone())
+	}
+	
+	#[setter]
+	fn set_name(&mut self, value: String) -> pyo3::PyResult<()> {
+		self.name = value;
+		Ok(())
+	}
+
+	fn __str__(&self) -> String {
+		format!("{self:?}")
+	}
 }
 
 #[pymethods]
 impl Config {
 	#[new]
-	#[pyo3(signature = (*, username, password, **kwds))]
+	#[pyo3(signature = (username, password, **kwds))]
```

Comment:
`username` and `password` definitely should be mandatory so this is not intended. Doesn't leading `*` mean that any number of positional arguments can be given before username/password? Can we make it so that those (and only those) are mandatory?

Reply:
@alemidev

Reply:
i see, making them positional OR keyword is surely desirable! if anyone can undo the change i'd love it <3 , otherwise i can fix it later today or tomorrow

Reply:
With the * in front they are kword only! 
Any positional argument gets simply ignored as it is swept under a nameless variable.
I can revert this later tonight probably, drowning in work atm

Reply:
> the Config::new() is weird and inconsistent across languages anyway

Yeah, the `Default` thing's a good idea, not sure how we didn't think of that. Java'll be left as the odd one out, but it kind of deserves it anyway.

```java
Config config = new Config("root", "toor");
Config config = new Config("root", "toor", "api.code.mp", 1234, true);
// no optional arguments?
```